tsinfer/inference.py

```diff
@@ -370,18 +419,21 @@ def augment_ancestors(
     ancestors_ts,
     indexes,
     *,
-    num_threads=0,
-    path_compression=True,
     recombination_rate=None,
-    mismatch_rate=None,
-    precision=None,
-    extended_checks=False,
-    engine=constants.C_ENGINE,
-    progress_monitor=None,
+    mismatch_ratio=None,
+    recombination=None,  # Deliberately undocumented: allows passing prob array directly
+    mismatch=None,  # Deliberately undocumented: allows passing prob array directly
+    path_compression=True,
+    # Performance related parameters below
```

Comment:
Ditto comments above - let's order the params by documented/undocumented, with a "# Following parameters are deliberately undocumented and may change arbitrarily in future versions".